src/olmo_core/train/trainer.py

```diff
@@ -743,10 +743,19 @@ def load_checkpoint(self, dir: PathOrStr, *, load_trainer_state: Optional[bool]
 
         # NOTE: to avoid making a ton of client requests (S3 or otherwise) we only make those
         # requests from rank 0 then scatter the result to the other ranks.
+        dir_to_scatter: Optional[PathOrStr] = dir
+        error: Optional[Exception] = None
         if get_rank() == 0 and not self.checkpointer.dir_is_checkpoint(dir):
-            # Try to find the latest checkpoint in the directory.
-            dir = self.checkpointer.latest_checkpoint(dir)
-        dir = scatter_object(dir)
+            try:
+                dir_to_scatter = self.checkpointer.latest_checkpoint(dir)
+            except FileNotFoundError as e:  # defer raising until after the scatter
+                dir_to_scatter, error = None, e
+        dir_to_scatter = scatter_object(dir_to_scatter)
```

Comment:
This avoids a 30 min hang (rank 0 throws exception while other ranks are waiting) when the checkpoint isn't found

Reply:
Oh, this is a great fix!